src/main/scala/com/netflix/edda/mongo/MongoDatastore.scala

```diff
@@ -365,7 +365,8 @@ class MongoDatastore(val name: String) extends Datastore {
 
   override def remove(queryMap: Map[String, Any])(implicit req: RequestId) {
     try {
-      primary.remove(mapToMongo(queryMap))
+      var opResult = primary.remove(mapToMongo(queryMap, true))
```

Comment:
It seems like the main change is to pass in true here. For posterity, can you describe how this fixes the issue? What did it do without that and how does this fix it?